src/mame/drivers/atarigt.cpp

```diff
@@ -412,12 +412,12 @@ void atarigt_state::primrage_protection_w(address_space &space, offs_t offset, u
 	primrage_update_mode(offset);
 
 	/* check for certain read sequences */
-	if (m_protmode == 1 && offset >= 0xdc7800 && offset < 0xdc7800 + sizeof(m_protdata) * 2)
-		m_protdata[(offset - 0xdc7800) / 2] = data;
+	if (m_protmode == 1 && offset >= 0xdc7800 && offset < 0xdc7800 + sizeof(m_protdata.get()) * 2)
```

Comment:
This won't work - `m_protdata.get()` returns a pointer, so you're just using the size of a pointer (four or eight bytes), not the size of the allocated block.